File: src/main/java/leetcode/editor/cn/MergeTwoSortedLists.java
package leetcode.editor.cn;

import model.ListNode;

/**
 * [21] 合并两个有序链表
 */
public class MergeTwoSortedLists {
	//leetcode submit region begin(Prohibit modification and deletion)

	/**
	 * Definition for singly-linked list. public class ListNode { int val; ListNode next; ListNode()
	 * {} ListNode(int val) { this.val = val; } ListNode(int val, ListNode next) { this.val = val;
	 * this.next = next; } }
	 */
	class Solution {

		public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
			ListNode dummy = new ListNode(-1);
			ListNode pointer = dummy;
			while (true) {
				if (list1 == null) {
					pointer.next = list2;
					break;
				}
				if (list2 == null) {
					pointer.next = list1;
					break;
				}
				if (list1.val > list2.val) {
					pointer.next = list2;
					list2 = list2.next;
				} else {
					pointer.next = list1;
					list1 = list1.next;
				}
				pointer = pointer.next;

			}
			return dummy.next;
		}
	}
//leetcode submit region end(Prohibit modification and deletion)

}

File: src/main/java/leetcode/editor/cn/MergeTwoSortedLists.go
  package leetcode.editor.cn;
  
  /**
   * [21] 合并两个有序链表
   */
  public class MergeTwoSortedLists{
      public static void main(String[] args) {
           Solution solution = new MergeTwoSortedLists().new Solution();
      }
      //leetcode submit region begin(Prohibit modification and deletion)
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func mergeTwoLists(list1 *ListNode, list2 *ListNode) *ListNode {
  dummy := &ListNode{}
  p := dummy
  for list1 != nil && list2 != nil{
    if list1.Val < list2.Val {
      p.Next = list1
      list1 = list1.Next
    } else {
      p.Next = list2
      list2 = list2.Next
    }
    p = p.Next
  }
  if list1 != nil {
    p.Next = list1
  }
  if list2 != nil {
    p.Next = list2
  }

  return dummy.Next
}
//leetcode submit region end(Prohibit modification and deletion)

  }